server/domain/devices/waterConsumption.go

```diff
@@ -0,0 +1,217 @@
+package devices
+
+import (
+	"fmt"
+	//"github.com/jinzhu/gorm"
+)
+
+// WaterConsumption ...
+type WaterConsumption struct {
+	Sensor
+	Events []WaterConsumptionEvent `gorm:"foreignkey:SensorType;association_foreignkey:Type"`
+}
+
+// WaterConsumptionEvent ...
+type WaterConsumptionEvent struct {
+	Event
+	Name        string `json:"name"`
+	Consumption float32
+}
+
+// TableName ...
+func (WaterConsumption) TableName() string {
+	return "sensors"
+}
+
+func init() {
+	fmt.Printf("Initalising %s sensor... \n", WaterMeter)
+}
+
+// Get ...
+func (wc *WaterConsumption) Get() (*WaterConsumption, error) {
+	device := new(WaterConsumption)
+	err := Storage.Where(&Sensor{Type: WaterMeter}).First(&device).Error
+	if err != nil {
+		// returning custom DB error message
+		err = ErrNotFound
+	}
+
+	return device, err
+}
+
+// Ping ...
+func (wc *WaterConsumption) Ping() (*[]WaterConsumptionEvent, error) {
```

Comment:
Is it used? If it's just for ping you don't need to return the whole bunch of events